.buildkite/scripts/common.sh

```diff
@@ -12,14 +12,14 @@ create_workspace() {
 
 install_go_dependencies() {
     local install_packages=(
-            "github.com/magefile/mage"
-            "github.com/elastic/go-licenser"
-            "golang.org/x/tools/cmd/goimports"
-            "github.com/jstemmer/go-junit-report"
-            "gotest.tools/gotestsum"
+            "github.com/magefile/mage@latest"
+            "github.com/elastic/go-licenser@latest"
+            "golang.org/x/tools/cmd/goimports@v0.24.0"
```

Comment:
Why this is the only one with a version and not latest? Additionally, I'm not sure if we should rely on a stable version to avoid unrelated breaking changes on CI runs.

Reply:
Nvm, I should have read the description fully. I still consider a good practice to pin versions on these installs as well, downside is that we have to update this periodically.

Reply:
Our .go-version is currently 1.21, and the latest version of goimports requires Go 1.22. This is being addressed separately in #110, so these changes should disappear from this PR once that one is merged.